source/docs/software/commandbased/commands-v3/telemetry-and-debugging.rst

```diff
@@ -0,0 +1,59 @@
+# Telemetry and Debugging
+
+.. todo::
+   This article covers using v3's enhanced telemetry features and debugging tools to understand and troubleshoot robot behavior. Good telemetry is essential for rapid iteration during competition, and v3 provides improved tools compared to v2.
+
+## v3 Telemetry Improvements
+
+.. todo::
+   Overview of telemetry enhancements in v3. Cover:
+   - What telemetry improvements v3 provides over v2
+   - Built-in command scheduler telemetry
+   - Mechanism state visibility
+   - Command lifecycle events
+   - Priority and interruption logging
+   - Performance metrics (cycle time, CPU usage)
+   - How telemetry integrates with the scheduler
+
+## Dashboard Integration
+
+.. todo::
+   Explain how to integrate v3 telemetry with dashboards. Cover:
+   - Supported dashboard tools (Shuffleboard, Glass, etc.)
+   - Automatic command scheduler widget/visualization
+   - Publishing Mechanism state to Network Tables
+   - Publishing command properties and status
+   - Creating custom dashboard layouts for v3 projects
+   - Live tuning parameters through the dashboard
+   - Remote command triggering from dashboard
+   - Best practices for dashboard organization
+
+## Debugging Techniques
+
+.. todo::
+   Provide techniques for debugging v3 command-based code. Cover:
+   - Using dashboard telemetry to diagnose issues
+   - Logging command lifecycle events
+   - Tracing command execution flow
+   - Debugging priority conflicts
+   - Debugging requirement conflicts
+   - Debugging trigger bindings that don't fire
+   - Debugging commands that don't end/interrupt properly
+   - Using simulation for debugging
+   - Breakpoint debugging in IDEs with v3
```

Comment:
This would have to be in sim or unit tests, no way are we going to recommend breakpoint debugging on running robots